src/stream-match.ts

```diff
@@ -14,26 +27,30 @@ export const streamMatchMessage = z
         .nonnegative()
         .int()
         .describe("the duration of the game in seconds"),
-      gameParams: z
-        .object({
-          level: z
-            .number()
-            .nonnegative()
-            .int()
-            .min(1)
-            .max(2)
-            .describe("game difficulty"),
-        })
-        .describe("to be confirmed for each game, should be provided by CIMU"),
+      gameParams,
     }),
     z.object({
       kind: z.literal("[host]:start-game"),
+      timeLeftInSeconds: z
+        .number()
+        .nonnegative()
+        .int()
+        .describe(
+          "how many seconds left before the game will end, should be 0 <= timeLeft <= gameDurationInSeconds",
```

Comment:
@leehong0704 Can we add this?

Reason is because we have a timer, StreamMatch also has a timer, and so there will be a case when user joins mid-way of our timer, but it will be fresh 60 seconds timer in your game, so this field is to override it